src/components/task-plan/builder/index.cjsx

```diff
@@ -146,42 +150,46 @@ TaskPlanBuilder = React.createClass
 
   setAllPeriods: ->
     {courseId} = @props
+    {showingPeriods, savedAllTaskings} = @state
 
     #save current taskings
-    if @state.showingPeriods
```

Comment:
yes, it wasnt really doing anything when it was in there, `@state.showingPeriods` was always true